src/nvim/buffer.c

```diff
@@ -1335,11 +1336,12 @@ void do_autochdir(void)
 static int top_file_num = 1;            /* highest file number */
 
 buf_T *
-buflist_new (
-    char_u *ffname,            /* full path of fname or relative */
-    char_u *sfname,            /* short fname or NULL */
-    linenr_T lnum,                  /* preferred cursor line */
-    int flags                      /* BLN_ defines */
+buflist_new(
+    char_u *ffname,            // full path of fname or relative
+    char_u *sfname,            // short fname or NULL
+    linenr_T lnum,             // preferred cursor line
+    int flags,                 // BLN_ defines
+    handle_T bufnr
```

Comment:
> Can we instead use the same ffname every time, so that buflist_new() finds the same existing incsub buffer?

No we wipe the buffer when we're done. If the buffer exists then the user could mess up the buffer by changing settings etc until the next time.

> Hmm, even q: and :copen use a new bufnr each time.

but "each time" is a lot more often for incsub, almost every keypress after `:%s`.